docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/components/usa-footer/usa-footer.ts

```diff
@@ -153,16 +153,29 @@ const initialize = (): void => {
 	if (document.documentElement.lang === 'es') {
 		NCIBigFooter.create(footerElement, {
 			subscribeInvalidEmailAlert: 'Ingrese su dirección de correo electrónico',
+			backToTopText: 'Volver Arriba',
 		});
 	} else {
 		NCIBigFooter.create(footerElement);
 	}
 
 	// set up listener for backToTop
-	const backToTopLink = footerElement.querySelector(
-		'.usa-footer__nci-return-to-top a'
-	) as HTMLAnchorElement;
-	backToTopLink.addEventListener('click', backToTopAnalyticsHandler());
+	const bttObserver = new MutationObserver(() => {
+		const selector = '.usa-footer__nci-return-to-top a';
+		if (document.querySelector(selector)) {
+			const backToTopLink = footerElement.querySelector(
+				selector
+			) as HTMLAnchorElement;
+			backToTopLink.addEventListener('click', backToTopAnalyticsHandler());
+			bttObserver.disconnect();
+		}
+	});
+	bttObserver.observe(document, {
+		attributes: false,
+		childList: true,
+		characterData: false,
+		subtree: true,
+	});
```

Comment:
This code is not very performant. A few items:
1. A `MutationObserver` will get a list of mutations passed in, and you can identify the specific return to top link through that. 
2. This code it scanning the document for selectors for every element, just testing elements would save time.
3. This code (at 173) is observing the entire document, which should be, `document.body` at the **very least** since we know the head will not have any of the dom elements. However, the return to top button is a child of the footer. So before initializing the footer javascript, you could setup the observer to look at just the footer element that is in the DOM. This would be substantially less elements that would have to be parsed.
    a. Furthermore, the RTT is a direct descendant of the footer.  So if you are observing the footer element, you can change `subtree: false` which means even less mutations to sift through.

The code would maybe be:
```
var observer = new MutationObserver(function(mutations, theObserver) {
  // Iterate through each change
  mutations.forEach(function(mutation) {
    // Now loop through the elements that have been added, looking for the
    // link
    mutation.addedNodes.forEach(function(node) {
      if (node is HTMLElement && node.matches(.usa-footer__nci-return-to-top a)) {
         node.addEventListener(...);
         theObserver.disconnect();
      }
    } 
  }
}
// Set the listener on the body.
observer.observe(footerElement, {
   childList: true, 
});
```

The only possible issue here, which should not matter for the immediate future, is if the footer removes the BTT link, then the click handler stays. (Which is a memory leak situation) However, https://github.com/NCIOCPL/ncids/issues/1923 should be fixed by that time.